src/com/b44t/messenger/rpc/Rpc.java

```diff
@@ -0,0 +1,120 @@
+package com.b44t.messenger.rpc;
+
+import com.b44t.messenger.DcJsonrpcInstance;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutionException;
+
+public class Rpc {
+    private Map<Integer, CompletableFuture<JsonElement>> requestFutures = new ConcurrentHashMap<>();
```

Comment:
I don't know, HashMap is used in a similar fashion, maybe @Hocuri can clarify

Reply:
Either way is fine.

> Casting to `Map` interface immediately only reduces the possibilities of usage as far as I understand

That's correct, but since we don't need these other possibilities of usage, in Java it's idiomatic to use the more general type (although we do not have to follow idiomatic Java though).

> like casting to a dynamic trait object in Rust.

Yes, although every variable in Java is actually what would be a dynamic trait object in Rust. So, we have the choice between `Box<dyn ConcurrentHashMap>` or `Box<dyn Map>` here.